packages/syft/tests/syft/dataset/dataset_stash_test.py

```diff
@@ -0,0 +1,82 @@
+# stdlib
+from typing import List
+
+# third party
+import pytest
+
+# syft absolute
+from syft.core.node.new.dataset import Dataset
+from syft.core.node.new.dataset_stash import ActionIDsPartitionKey
+from syft.core.node.new.dataset_stash import NamePartitionKey
+from syft.core.node.new.document_store import QueryKey
+from syft.core.node.new.document_store import QueryKeys
+from syft.core.node.new.uid import UID
+
+
+def test_dataset_namepartitionkey():
+    name_partition_key = QueryKeys(qks=NamePartitionKey).all[0]
```

Comment:
To generate a Querykey from PartitionKey we can either:
`QueryKey.from_obj(partition_key=NamePartitionKey, obj='Dummy Name')`
or call
name_query_key = NamePartitionKey.with_obj('Dummy Name')  # pass in the value directly for the partition key
Or
@dataclass
class MockObj:
    name: str

obj = MockObj(name="Dummy Name")
name_query_key = NamePartitionKey.with_obj(obj)  # pass an object which has partition key as one of its class members.

Note: A QueryKey is has the attributes of the PartitionKey plus `value`, which holds the value.

You can refer to this test for [reference](https://github.com/OpenMined/PySyft/pull/7462/files#diff-39617657524bd6b2b22dfb196f13b928a252c0f4fd2ed76e5584bf7c85d996f3R82)